From a6480aca3e0e8c0509c852dae89d2016b616c486 Mon Sep 17 00:00:00 2001
From: Aloka Dixit <quic_alokad@quicinc.com>
Date: Fri, 2 Jul 2021 12:21:11 -0700
Subject: [PATCH 11/23] eht: parse elements received in management frames

Parse and store pointers to EHT capabilities and operation elements
received in management frames.

Signed-off-by: Aloka Dixit <quic_alokad@quicinc.com>
---
 src/common/ieee802_11_common.c | 8 ++++++++
 src/common/ieee802_11_common.h | 4 ++++
 2 files changed, 12 insertions(+)

--- a/src/common/ieee802_11_common.c
+++ b/src/common/ieee802_11_common.c
@@ -311,6 +311,14 @@ static int ieee802_11_parse_extension(co
 		elems->mbssid_known_bss = pos;
 		elems->mbssid_known_bss_len = elen;
 		break;
+	case WLAN_EID_EXT_EHT_CAPABILITY:
+		elems->eht_capabilities = pos;
+		elems->eht_capabilities_len = elen;
+		break;
+	case WLAN_EID_EXT_EHT_OPERATION:
+		elems->eht_operation = pos;
+		elems->eht_operation_len = elen;
+		break;
 	default:
 		if (show_errors) {
 			wpa_printf(MSG_MSGDUMP,
--- a/src/common/ieee802_11_common.h
+++ b/src/common/ieee802_11_common.h
@@ -119,6 +119,8 @@ struct ieee802_11_elems {
 	const u8 *s1g_capab;
 	const u8 *pasn_params;
 	const u8 *mbssid_known_bss;
+	const u8 *eht_capabilities;
+	const u8 *eht_operation;
 
 	u8 ssid_len;
 	u8 supp_rates_len;
@@ -174,6 +176,8 @@ struct ieee802_11_elems {
 	u8 sae_pk_len;
 	u8 pasn_params_len;
 	u8 mbssid_known_bss_len;
+	u8 eht_capabilities_len;
+	u8 eht_operation_len;
 
 	struct mb_ies_info mb_ies;
 	struct frag_ies_info frag_ies;
